src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -1211,6 +1211,75 @@ public void testGenerateXpathSuppressionOptionTwo(@SysErr Capturable systemErr,
+                + "VARIABLE_DEF[./IDENT[@text='a']]\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/"
+                + "OBJBLOCK/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/"
+                + "LITERAL_FOR\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/"
+                + "OBJBLOCK/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/"
+                + "LITERAL_FOR/SLIST/EXPR/PLUS_ASSIGN/IDENT[@text='i']\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/"
+                + "OBJBLOCK/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/"
+                + "LITERAL_FOR/SLIST/RCURLY\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/OBJBLOCK"
+                + "/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/LITERAL_IF\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/OBJBLOCK"
+                + "/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/LITERAL_IF/SLIST/"
+                + "LITERAL_RETURN\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/OBJBLOCK"
+                + "/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/LITERAL_IF/SLIST/"
+                + "RCURLY\"/>",
+            "<suppress-xpath",
+            "       files=\"InputMainGenerateXpathIndentation.java\"",
+            "       checks=\"IndentationCheck\"",
+            "       query=\"/CLASS_DEF[./IDENT[@text='InputMainGenerateXpathIndentation']]/OBJBLOCK"
+                + "/METHOD_DEF[./IDENT[@text='generateIndentationXpath']]/SLIST/RCURLY\"/>",
+            "</suppressions>"
+        );
+
+        Main.main("-c", "/google_checks.xml", "--generate-xpath-suppression",
+                getPath("InputMainGenerateXpathIndentation.java"));
```

Comment:
please explain a reason of this test and this extra commit. Why we need this test in MainTest ?

Reply:
Actually,
https://github.com/checkstyle/checkstyle/blob/c42a0e1b63c33400e05c98b62179cf303724e525/src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java#L1081
This method above got overpopulated due to generation of indentation x-path as google check was expecting 2 tab spaces instead of 4.
Thus I changed [InputMainComplexityOverflow.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/main/InputMainComplexityOverflow.java).

>  Why we need this test in MainTest ?

After alterations there were no MainTest that was checking generated indentation x-path so I thought adding that.

> if it is to resolve pitest mutation, please clearly state this in commit message.

I have checked the pitest mutation and maven verification; removing this test doesn't cause any change or mutation violations.

So I cant say that they are important or needed anymore.

Should I remove that now.

Reply:
Please remove this commit, MainTest is only care of verification of Main.java file. All xpath in Indentation should be at XpathRegressionIndentationTest , if you think that something is missing at XpathRegressionIndentationTest please extend it.